src/grammar.coffee

```diff
@@ -597,8 +601,22 @@ grammar =
   # where only values are accepted, wrapping it in parentheses will always do
   # the trick.
   Parenthetical: [
-    o '( Body )',                               -> new Parens $2
-    o '( INDENT Body OUTDENT )',                -> new Parens $3
+    o '( Body )',                               ->
+        # When `Parens` block includes a `StatementLiteral` (e.g. `(b; break) for a in arr`),
+        # it won't compile since `Parens` (`(b; break)`) is compiled as `Value` and
+        # pure statement (`break`) can't be used in an expression.
+        # For this reasons, we return `Block`, which is passed further to `Value` and `Expression`.
+        stm = $2.contains (n) -> n.constructor.name is 'StatementLiteral'
```

Comment:
I'd rather we don't add such code to the grammar. We can have a node function for that.

Reply:
Also we try not to use `constructor.name` for obfuscation reasons IIRC.